### PR TITLE
automation, Remove unneeded build repetitions

### PR DESCRIPTION
**Pull request by oshoval**
make `bazel-build-images` happens 4 times, even if all the times
succeeded, wasting resources.
Fix it by stopping when it first managed to finish successfully.

Signed-off-by: Or Shoval <oshoval@redhat.com>

```release-note
None
```

### CODE REVIEW

File: automation/test.sh [hunk 1]
```diff
@@ -173,13 +173,19 @@ export IMAGE_PREFIX_ALT=${IMAGE_PREFIX_ALT:-kv-}
 # probably because load on the node, possible situation when the bazel
 # fails to download artifacts, to avoid job fails because of it,
 # we repeat the build images action
+tries=3
```
Comment (EdDev) [inline automation/test.sh:176]: Can you extract this to a function?
The return value should hopefully be either 0 for success or the last attempt result.

**Review by EdDev — CHANGES_REQUESTED** [review]
Nice catch.

**Comment by oshoval** [comment]
Addressed comments.

File: automation/test.sh [hunk 2]
```diff
@@ -151,6 +151,24 @@ collect_debug_logs() {
...
+    local tries=3
+    for i in $(seq 1 $tries); do
+        if make bazel-build-images; then
+            return 0
+        else
+            rc=$?
+            if [ $i -eq $tries ]; then
+                return $rc
+            fi
+        fi
```
Comment (EdDev) [inline automation/test.sh:168]: Will this work?
```
for i in $(seq 1 3); do
    make bazel-build-images && return
    rc=$?
done

return $rc
```

Comment (oshoval) [reply automation/test.sh:168]: we have `-e` so i am not sure it will work, i can try if you think otherwise

**Comment by oshoval** [comment]
Addressed comments.

**Comment by oshoval** [comment]
Asking myself, what will happen if we will remove it totally,
as cluster-up / cluster-sync is enough usually.
I will try that on other PR

**Review by EdDev — APPROVED** [review]
Thanks.

**Comment by oshoval** [comment]
Thanks Edy

(failing tests are unreleated)

/cc @enp0s3

**Pull request merged**